### ISSUE TITLE
Drop-in 4.5.0 and 4.2.0 Unable to start activity crash with GooglePay

### ISSUE THREAD

**Issue by akashraje** [Adyen/adyen-android#632]
**Describe the bug**
Hi,
I am getting this crash when I call `DropIn.startPayment()`

```
403-22 16:23:32.246 11470 11470 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.coolapp.android.lab/com.adyen.checkout.dropin.ui.DropInActivity}: y5.b: ModelObject protocol requires a ModelObject.Serializer object called SERIALIZER on class g
503-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4031)
603-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4197)
703-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
803-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
903-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
1003-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2434)
1103-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Handler.dispatchMessage(Handler.java:106)
1203-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Looper.loopOnce(Looper.java:226)
1303-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Looper.loop(Looper.java:313)
1403-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.ActivityThread.main(ActivityThread.java:8633)
1503-22 16:23:32.246 11470 11470 E AndroidRuntime:    at java.lang.reflect.Method.invoke(Native Method)
1603-22 16:23:32.246 11470 11470 E AndroidRuntime:    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
1703-22 16:23:32.246 11470 11470 E AndroidRuntime:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1133)
1803-22 16:23:32.246 11470 11470 E AndroidRuntime: Caused by: y5.b: ModelObject protocol requires a ModelObject.Serializer object called SERIALIZER on class g
1903-22 16:23:32.246 11470 11470 E AndroidRuntime:    at com.adyen.checkout.core.model.ModelUtils.readModelSerializer(ModelUtils.java:7)
2003-22 16:23:32.246 11470 11470 E AndroidRuntime:    at com.adyen.checkout.core.model.ModelUtils.deserializeModel(ModelUtils.java:1)
2103-22 16:23:32.246 11470 11470 E AndroidRuntime:    at com.adyen.checkout.core.model.ModelObject$Creator.createFromParcel(ModelObject.java:3)
2203-22 16:23:32.246 11470 11470 E AndroidRuntime:    at com.adyen.checkout.core.model.ModelObject$Creator.createFromParcel(ModelObject.java:1)
2303-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Parcel.readParcelable(Parcel.java:3345)
2403-22 16:23:32.246 11470 11470 E AndroidRuntime:    at p6.c.<init>(GooglePayConfiguration.java:33)
2503-22 16:23:32.246 11470 11470 E AndroidRuntime:    at p6.c$a.createFromParcel(GooglePayConfiguration.java:1)
2603-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Parcel.readParcelable(Parcel.java:3345)
2703-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Parcel.readValue(Parcel.java:3238)
2803-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Parcel.readMapInternal(Parcel.java:3618)
2903-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Parcel.readHashMap(Parcel.java:2641)
3003-22 16:23:32.246 11470 11470 E AndroidRuntime:    at e6.f.<init>(DropInConfiguration.kt:2)
3103-22 16:23:32.246 11470 11470 E AndroidRuntime:    at e6.f$b.createFromParcel(DropInConfiguration.kt:2)
3203-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Parcel.readParcelable(Parcel.java:3345)
3303-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Parcel.readValue(Parcel.java:3238)
3403-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.Parcel.readArrayMapInternal(Parcel.java:3635)
3503-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
3603-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.BaseBundle.unparcel(BaseBundle.java:236)
3703-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.os.BaseBundle.getBoolean(BaseBundle.java:935)
3803-22 16:23:32.246 11470 11470 E AndroidRuntime:    at com.adyen.checkout.dropin.ui.DropInActivity.onCreate(DropInActivity.kt:8)
3903-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.Activity.performCreate(Activity.java:8282)
4003-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.Activity.performCreate(Activity.java:8262)
4103-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
4203-22 16:23:32.246 11470 11470 E AndroidRuntime:    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4005)
4303-22 16:23:32.246 11470 11470 E AndroidRuntime:    ... 12 more
```

**To Reproduce**
1. Include GooglePayConfiguration with `MerchantInfo` through `GooglePayConfiguration.Builder`
2. Enable proguard

**Smartphone (please complete the following information):**
 - Device: Pixel 2 XL
 - OS: Android 11
 - Browser NA
 - Version Adyen drop-in 4.5.0 and drop-in 4.2.0

**Additional context**

- This was tested only on `Drop-In 4.2.0` and `Drop-In 4.5.0`
- I noticed that this happens only on release builds where proguard is enabled.
- This happens only when I include GooglePayConfiguration.
- I found this happens only when I include `MerchantInfo` in `GooglePayConfiguration`.

It works fine when I remove `MerchantInfo` from `GooglePayConfiguration.Builder` but I would like to show the merchant name on googlepay payments.

This is how I am creating GooglePayConfigurations and DropIn.

```
val environment = if (BuildConfig.IS_LAB || BuildConfig.IS_UITEST ) Environment.TEST else Environment.LIVE

val dropInConfigBuilder = DropInConfiguration.Builder(
            context,
            PaymentDropInService::class.java,
            BuildConfig.ADYEN_CLIENT_KEY
        ).setAmount(amount)
            .setEnvironment(environment)

val googleMerchantInfo = MerchantInfo()
            .apply {
            merchantName = "Cool Merchant"
            merchantId = "Cool merchant's Id"
        }

val googlePayConfiguration = GooglePayConfiguration.Builder(context, BuildConfig.ADYEN_CLIENT_KEY)
                .setCountryCode(request.countryCode)
                .setAmount(amount)
                .setBuilderMerchantInfo(googleMerchantInfo) //Comment this out and everything works
                .setEnvironment(environment)
                .setMerchantAccount(merchantAccount)
                .build()

dropInConfigBuilder.addGooglePayConfiguration(googlePayConfiguration)

DropIn.startPayment(activity, dropInLauncher, paymentMethodsApiResponse, dropInConfigBuilder.build())

```

**Comment by jreij** [comment]
Hi, thanks for reaching out! This seems like a proguard issue, I assume it's only happening on release builds, right?

If so can you try to add this line to your proguard file (usually `proguard-rules.pro`:
```
-keep class com.adyen.checkout.googlepay.model.** { *; }
```

Let me know if that fixes the issue :)

**Comment by akashraje** [comment]
Thanks! **This works.**
I was initially trying with `com.adyen.checkout.base.model.** { *; }` which didn't work.

**Comment by jreij** [comment]
Glad it worked! In all cases we also solved this issue on our side but updating our proguard files so for the next release you can remove all your proguard rules related to our SDK and test again, it should work 😉

**Comment by jreij** [comment]
Hi again @akashraje , we released `4.6.0` last week and starting this version you should be able to use the SDK without manually adding any custom Proguard rules.
